testsuite/gnat2goto/tests/constant_declarations/test.out

```diff
@@ -1,2 +1,7 @@
+Error from cbmc test:
+function `__CPROVER_Ada_Overflow_Check' in module `' is shadowed by a definition in module `'
+function `__range_check5' in module `' is shadowed by a definition in module `'
+
```

Comment:
???

Reply:
It's a warning from `linking.cpp`: when we try to merge the two symbol tables they both contain functions with the same name. And if their types match, CBMC only emits this warning and throws away the _new_ definition. Which is ok for `__CPROVER_Ada_Overflow_Check` but maybe not for the `__range_checkn` which might be unique. I think we should mangle the file-name into the range-check function name somehow.

Reply:
I'm not sure I understand, where is the `__range_checkn` function coming from?  If we are generating it why isn't it `__CPROVER_Ada_Range_Check` or similar?

Reply:
Well `__range_check` is something you can use inside an expression so it has to evaluate to the same type as the expression you are range-checking. It then calls `__CPROVER_Ada_Check` which has type `void (int, char*)`.

Reply:
Right; this is a little bit concerning because it suggests that when we link things we risk using the wrong range check.  Maybe the range check function should include the type so that they are unique.

( Minor point : maybe is should call `__CPROVER_Ada_Range_Check`)